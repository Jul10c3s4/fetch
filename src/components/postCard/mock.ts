//geralmente cria-se um mock quando o componente é estático para que o mock seja usado tanto em tests quanto no próprio componente

export const postCardPropsMock = {
    title: "titulo",
    body: "corpo",
    id: 1,
    photo: "https://dfdfdf"
}