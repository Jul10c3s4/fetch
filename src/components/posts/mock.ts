export const postsCardsPropsMocks = {
    posts:[
      {
        id: 1,
        title: 'title 1',
        body: 'body 1',
        photo: 'https://dfdf'
      },
      {
        id: 2,
        title: 'title 2',
        body: 'body 2',
        photo: 'https2://dfdf'
      },
      {
        id: 3,
        title: 'title 3',
        body: 'body 3',
        photo: '3https://dfdf'
      },
      {
        id: 4,
        title: 'title 4',
        body: 'body 4',
        photo: '4https://dfdf'
      },
    ]
}