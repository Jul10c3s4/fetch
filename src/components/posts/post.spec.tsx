import { render, screen } from '@testing-library/react';
import { Posts } from '.';
import '@testing-library/jest-dom/extend-expect';
import { postsCardsPropsMocks } from './mock';

const props = postsCardsPropsMocks;
describe('<Posts>', () => {
  it('should render posts', () => {
    render(<Posts {...props} />);
    expect(screen.getAllByRole('heading', { name: /title/i })).toHaveLength(4);

    expect(screen.getAllByRole('img', { name: /title/i })).toHaveLength(4);
    expect(screen.getByRole('img', { name: /title 1/i })).toHaveAttribute('src', 'https://dfdf');
    expect(screen.getAllByText(/body/i)).toHaveLength(4);
  });

  it('should render posts', () => {
    const { container } = render(<Posts {...props} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
