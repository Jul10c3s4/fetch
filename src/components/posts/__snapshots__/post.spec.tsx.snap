// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts> should render posts 1`] = `
<div
  class="posts"
>
  <div
    class="post"
  >
    <img
      alt="title 1"
      src="https://dfdf"
    />
    <div
      class="post-content"
    >
      <h1>
        1
      </h1>
      <h2>
        title 1
      </h2>
      <p>
        body 1
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="title 2"
      src="https2://dfdf"
    />
    <div
      class="post-content"
    >
      <h1>
        2
      </h1>
      <h2>
        title 2
      </h2>
      <p>
        body 2
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="title 3"
      src="3https://dfdf"
    />
    <div
      class="post-content"
    >
      <h1>
        3
      </h1>
      <h2>
        title 3
      </h2>
      <p>
        body 3
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="title 4"
      src="4https://dfdf"
    />
    <div
      class="post-content"
    >
      <h1>
        4
      </h1>
      <h2>
        title 4
      </h2>
      <p>
        body 4
      </p>
    </div>
  </div>
</div>
`;
